src/components/utils.js

```diff
@@ -37,3 +37,37 @@ export function isRunning (status: string): boolean {
   console.log('---------- status: ', status)
   return ['wait_for_launch', 'up', 'powering_up', 'powering_down', 'migrating', 'paused'].includes(status)
 }
+
+export function getFilteredClusters (clusters: Array<Object>, config: Object): Array<Object> {
+  const userId = config.getIn(['user', 'id'])
+  const allowedPermissions = [
+    'VmCreator',
+    'PowerUserRole',
+    'ClusterAdmin',
+    'DataCenterAdmin',
+    'UserVmManager',
+    'UserVmRunTimeManager',
+    'SuperUser',
+  ]
+  return clusters.filter((cluster: Object): boolean => {
+    const filteredRoles = cluster.get('permissions').filter((role) => (
+      role.get('userId') === userId && allowedPermissions.includes(role.get('name'))
+    ))
+    return !!filteredRoles.size
+  })
+}
+
+export function doesUserCanEditVm (vm: VmType, config: Object): boolean {
```

Comment:
Also, you could attach this to the VM redux object in the VM `toInternal` transform, so it is only done once.

It would require permissions to be presented and therefore a non-shallow fetch. Since refresh and VM card list pages both use shallow fetches, the app can't know a VM is editable by its user until the user opens a detail page and a deep fetch is done.  Maybe a shallow fetch needs to be just a little bit deeper?